Field.php

```diff
@@ -2,261 +2,96 @@
-                {
-                    $neighbours++;
-                }
-            }
-        }
-        return $neighbours;
+        return $this->y;
     }
 
     /**
-     * Calculate next generation
-     * First $nextField is created, which copies the contents of the $fields variable and sets the cell-status to 0 (dead).
-     * After that, the "set-alive-conditions" will be implemented and cells fitting the conditions will be set to 1 (alive).
+     * Count the number of living neighbours.
+     * @return int
      */
-    private function nextGeneration()
+    public function numberOfLivingNeighbours()
     {
-        $nextField = [];
-
-        //set nextField to 0
-        for ($y = 0; $y < $this->height; $y++)
-        {
-            for ($x = 0; $x < $this->width; $x++)
-            {
-                $nextField[$x][$y] = 0;
-            }
-        }
-
-        for ($y = 0; $y < $this->height; $y++)
-        {
-            for ($x = 0; $x < $this->width; $x++)
-            {
-                $neighbourCount = $this->countNeighbours($x, $y);
-
-                if ($neighbourCount === 3)
-                {
-                    //set alive
-                    $nextField[$x][$y] = 1;
-                }
-
-                if ($neighbourCount == 2 and $this->field[$x][$y] == 1)
-                {
-                    $nextField[$x][$y] = 1;
-                }
-            }
-        }
-
-        $this->field = $nextField;
+        return $this->board->getNeighboursOfField($this);
     }
 
     /**
-     * Start-Game function
-     * Prints the field, calculates the next Generation and prints the new field for "maxSteps"-times.
-     * Also, if a field is printed double and the field does'nt change anymore, the program will stop automatically.
-     * @param $_output BaseOutput The Output that should be used to output the field.
+     * Count the number of dead neighbours.
+     * @return int
      */
-    public function start($_output)
+    public function numberOfDeadNeighbours()
     {
-        $_output->outputField($this);
-
-        for ($i = 0; $i < $this->maxSteps; $i++)
-        {
-            $this->history[] = $this->field; //saves field
-            $this->generationCount++;
-            $this->nextGeneration();
-            $_output->outputField($this);
-
-            //compare
-            foreach ($this->history as $previousField)
-            {
-                $equal = true;
-                for ($y = 0; $y < $this->height; $y++)
-                {
-                    for ($x = 0; $x < $this->width; $x++)
-                    {
-                        if ($previousField[$x][$y] != $this->field[$x][$y])
-                        {
-                            $equal = false;
-                        }
-                    }
-                }
-                if ($equal == true)
-                {
-                    return;
-                }
-            }
+        $deadNeighbours = 8;
```

Comment:
This is wrong. The number of dead neighbours is not 8! This is the number of neighbours that a field may have. So either rename this to `$numberOfNeighbours` or just simply change the return statement below to
`return 8 - $this->numberOfLivingNeighbours();`